common/sketch/src/main/java/org/apache/spark/util/sketch/BloomFilter.java

```diff
@@ -195,13 +195,14 @@ private static int optimalNumOfHashFunctions(long n, long m) {
   /**
    * Computes m (total bits of Bloom filter) which is expected to achieve, for the specified
    * expected insertions, the required false positive probability.
-   *
-   * See http://en.wikipedia.org/wiki/Bloom_filter#Probability_of_false_positives for the formula.
+   * <p>
+   * See <a href="http://en.wikipedia.org/wiki/Bloom_filter#Probability_of_false_positives">
+   *     http://en.wikipedia.org/wiki/Bloom_filter#Probability_of_false_positives</a> for the formula.
    *
    * @param n expected insertions (must be positive)
    * @param p false positive rate (must be 0 < p < 1)
    */
-  private static long optimalNumOfBits(long n, double p) {
+  public static long optimalNumOfBits(long n, double p) {
```

Comment:
Change to `public` is because `DataFrameStatFunctions#buildBloomFilter` needs to use this method to calculate the `numBits` from `expectedNumItems` and `fpp`

Reply:
If you find `(must be 0 &lt; p &lt; 1)` to be quite messy, we can try changing it to `(must be {@literal 0 < p < 1})`

Reply:
I am good.